x-pack/plugin/mapper-constant-keyword/src/test/java/org/elasticsearch/xpack/constantkeyword/mapper/SearchIdleTests.java

```diff
@@ -0,0 +1,319 @@
+        assertEquals(RestStatus.OK, client().admin().indices().prepareRefresh(idleIndex, activeIndex).get().getStatus());
+
+        waitUntil(
+            () -> Arrays.stream(client().admin().indices().prepareStats(idleIndex, activeIndex).get().getShards())
+                .allMatch(ShardStats::isSearchIdle),
+            2,
+            TimeUnit.SECONDS
+        );
+
+        final IndicesStatsResponse idleIndexStatsBefore = client().admin().indices().prepareStats("test1").get();
+        assertIdleShard(idleIndexStatsBefore);
+
+        final IndicesStatsResponse activeIndexStatsBefore = client().admin().indices().prepareStats("test2").get();
+        assertIdleShard(activeIndexStatsBefore);
+
+        // WHEN
+        final SearchResponse searchResponse = client().prepareSearch("test*")
+            .setQuery(new WildcardQueryBuilder("constant_keyword", "test2*"))
+            .setPreFilterShardSize(5)
+            .get();
+
+        // THEN
+        assertEquals(RestStatus.OK, searchResponse.status());
+        assertEquals(idleIndexShardsCount, searchResponse.getSkippedShards());
+        assertEquals(0, searchResponse.getFailedShards());
+        Arrays.stream(searchResponse.getHits().getHits()).forEach(searchHit -> assertEquals("test2", searchHit.getIndex()));
+        final IndicesStatsResponse idleIndexStatsAfter = client().admin().indices().prepareStats(idleIndex).get();
+        assertIdleShardsRefreshStats(idleIndexStatsBefore, idleIndexStatsAfter);
+    }
+
+    private SearchResponse search(final String index, final String field, final String value, int preFilterShardSize) {
+        return client().prepareSearch(index)
+            .setQuery(new MatchPhraseQueryBuilder(field, value))
+            .setPreFilterShardSize(preFilterShardSize)
+            .get();
+    }
+
+    private static void assertIdleShard(final IndicesStatsResponse statsResponse) {
+        Arrays.stream(statsResponse.getShards()).forEach(shardStats -> assertTrue(shardStats.isSearchIdle()));
+        Arrays.stream(statsResponse.getShards()).forEach(shardStats -> assertTrue(shardStats.getSearchIdleTime() >= 100));
+    }
+
+    private static void assertIdleShardsRefreshStats(final IndicesStatsResponse before, final IndicesStatsResponse after) {
+        assertNotEquals(0, before.getShards().length);
+        assertNotEquals(0, after.getShards().length);
+        final List<RefreshStats> refreshStatsBefore = Arrays.stream(before.getShards())
+            .map(shardStats -> shardStats.getStats().refresh)
+            .toList();
+        final List<RefreshStats> refreshStatsAfter = Arrays.stream(after.getShards())
+            .map(shardStats -> shardStats.getStats().refresh)
+            .toList();
+        assertEquals(refreshStatsBefore.size(), refreshStatsAfter.size());
+        assertTrue(refreshStatsAfter.containsAll(refreshStatsBefore));
+        assertTrue(refreshStatsBefore.containsAll(refreshStatsAfter));
+    }
```

Comment:
It is odd that we don't have unit tests, yet I see that SearchServiceTests that already exists is also an `ESSingleNodeTestCase`. You may want to simplify your test taking inspiration from it, in that it pulls the search service from the node and calls canMatch directly against it.